### PR TITLE
PIMS-1260: React API Hooks

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

  
[PIMS-1260: ](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-1260)

I wound up implementing more stuff than expected, here's the quick rundown on what each piece is meant for.

* usePimsApi is where we will implement all the sub-hooks that define the various possible api operations. In those sub-hooks make the actual calls.
* useFetch is a wrapper around the native fetch method. With this hook you can make various http requests in a pretty similar style to axios, with a base url configured by the env and everything. Probably won't be using this much on it's own outside of the api hooks.
* useAsync basically buffers any async request in a component, ensuring that you don't fire another call if the same call is already pending. Without a mechanism like this you would end up making multiple duplicate requests thanks to re-renders of the component.
* useDataLoader is just there to sorta bundle up useAsync and useFetch into one convenient package. Pretty easy to request data automatically on page load or trigger it with a button or similar. See some examples of usage on the DevZone page. You can temporarily modify your getAccessRequestLatest in the express-api to experiment a bit.
* configContext for getting env variables and making them available wherever.
* authContext for getting the auth state, this could maybe be paired down to just using the keycloak package's version of this but I think eventually we will want to expand the context to contain users/roles info from the db here too.
* AuthRouteGuard is there to prevent you from going place that you shouldn't go without hitting keycloak first. I imagine we will use this a lot. Could extend to similar versions for specific roles and the like as well.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

**Review by LawrenceLau2020 — APPROVED** [review]
Looks good! I tested the "getUserAccessRequestLatest" endpoint where the api retrieves the latest access request from the database and it's retrieving the most recent access request based on my Keycloak user Id. @dbarkowsky you may want to have a look/test when you get a chance as well.

**Review by dbarkowsky — CHANGES_REQUESTED** [review]
Happy with the code. 
Would be good to have some docstrings for these functions/components though.

File: react-app/src/hooks/useDataLoader.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      if (!isMounted) {
+        return;
+      }
+      setData(response);
+    } catch (e) {
+      if (!isMounted) {
+        return;
+      }
+      setError(error);
+      errorHandler?.(error);
```
[the remark this answers is not in the document]
Comment (GrahamS-Quartech) [reply react-app/src/hooks/useDataLoader.ts:36]: Yup, it's super handy. C# also implements this nearly identically.

**Comment by GrahamS-Quartech** [comment]
Added some doc-strings / clarifying comments 👍

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**